src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -562,6 +564,76 @@ codet java_bytecode_convert_methodt::get_array_bounds_check(
 
 /*******************************************************************\
 
+Function: throw_arithmetic_exception
+
+  Inputs:
+
+ Outputs:
+
+ Purpose: Checks whether there is a division by zero 
+          and throws ArithmeticException if necessary
+
+\*******************************************************************/
+
+codet java_bytecode_convert_methodt::throw_arithmetic_exception(
```

Comment:
Consider adding an annotation here and then moving this to your existing instrumentation pass